### PR TITLE
[Fix] Upgraded the packages with the latest version

### PR DESCRIPTION
**Pull request by shubhdevelop**
Added a dev Dependency as fallback, there was a
packages babel-preset-react-app that was directly importing @babel/plugin-proposal-private-property-in-object, without defining the package.json, but was but was present in node_module for some other purposes, it could cause break in future,

This was a bug as create-react-app is no longer being maintained, thus it was unlikely the bug was going to be fixed.

Date:  20 Aug 2024

Developer Name: Shubham Raj

---

## Issue Ticket Number

#55 

Update the package in the Repo to current latest version, ensuring there is no breaks, and all the dependencies are working fine. Tests are are working as intended.
### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag
- [ ] Yes
- [x] No

### Database Changes
- [ ] Yes
- [x] No

- Dev server is also working fine
### Breaking Changes

- [x] Yes
- [ ] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Build Working</summary>

image

</details>

## Test Coverage
<details>
<summary>Test Coverage</summary>

image

</details>

### Dev Server
[screenshare](https://github.com/user-attachments/assets/878ec543-6a9d-4b8a-a0de-49f8c35f54ff)

### CODE REVIEW

**Comment by akankshadharkar** [comment]
Why does this PR add a component?

File: package.json [hunk 1]
```diff
@@ -3,13 +3,14 @@
...
-    "react-dom": "^18.2.0",
+    "@testing-library/dom": "^10.4.0",
+    "@testing-library/jest-dom": "^6.4.8",
+    "@testing-library/react": "^16.0.0",
+    "@testing-library/user-event": "^14.5.2",
+    "react": "^18.3.1",
+    "react-dom": "^18.3.1",
     "react-scripts": "5.0.1",
-    "web-vitals": "^3.1.0"
+    "web-vitals": "^4.2.3"
```
Comment (iamitprakash) [inline package.json]: can we remove ^ from all where

**Review by iamitprakash — CHANGES_REQUESTED** [review]

**Comment by shubhdevelop** [comment]
> Why does this PR add a component?

App was breaking at the App.js while running tests, the component was imported but and no such component was found in index.js.

There was no such file as "./index.js" in that directory

File: package.json (hunk 1 above)
Comment (shubhdevelop) [reply package.json]: Yes we can

**Review by iamitprakash — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -3,13 +3,14 @@
...
   "private": true,
   "dependencies": {
-    "@testing-library/jest-dom": "^5.16.5",
-    "@testing-library/react": "^13.4.0",
-    "@testing-library/user-event": "^14.4.3",
-    "react": "^18.2.0",
-    "react-dom": "^18.2.0",
+    "@testing-library/dom": "10.4.0",
+    "@testing-library/jest-dom": "6.4.8",
+    "t @testing-library/react": "16.0.0",
```
Comment (ankushdharkar) [inline package.json:8]: Is this name correct?

File: package.json [hunk 3]
```diff
@@ -38,5 +39,8 @@
   "volta": {
     "node": "18.12.1",
     "yarn": "1.22.19"
+  },
+  "devDependencies": {
+    "@babel/plugin-proposal-private-property-in-object": "7.21.11"
```
Comment (ankushdharkar) [inline package.json:44]: Why do we need this?

File: src/components/starbucks-input/StarbucksInput.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export default function StartbucksInput() {
```
Comment (ankushdharkar) [inline src/components/starbucks-input/StarbucksInput.js:1]: Typo in *Starbucks name

File: src/components/starbucks-input/index.spec.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
-import StarbucksInput from './index.js';
+import StarbucksInput from './StarbucksInput.js';
```
Comment (ankushdharkar) [inline src/components/starbucks-input/index.spec.js:1]: Going to comment this out and let people create the file in their tests :)

**Review by ankushdharkar — COMMENTED** [review]
@iamitprakash @shubhdevelop There are issues in installing the packages after this PR. Please validate the changes next time before merging.

**Pull request merged**